--- a/echo.cabal	2019-05-04 19:57:08.114737098 -0400
+++ a/echo.cabal	2019-05-04 19:57:16.185772745 -0400
@@ -39,7 +39,7 @@
                        System.IO.Echo.Internal
 
   build-depends:       base    >= 4.3     && < 5
-                     , process >= 1.0.1.1 && < 1.5
+                     , process >= 1.0.1.1 && < 1.7
   if os(windows)
     cpp-options:       "-DWINDOWS"
     build-depends:     mintty >= 0.1 && < 0.2
