--- squashed-portage	2013-07-20 04:43:06.000000000 -0500
+++ /etc/init.d/squashed-portage	2014-08-24 15:18:24.578224853 -0500
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/sbin/runscript
 
 # Copyright 1999-2013 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
@@ -22,13 +22,7 @@
 # 2009-11-22, v.0.2.0(3) morris <mauricioarozi@gmail.com>
 # 2013-07-19, v.0.3.0(1) Andrey Ovcharov <sudormrfhalt@gmail.com>
 
-opts="sync"
-
-# lib
-[ -f /usr/lib/misc/core-functions.sh ] && source /usr/lib/misc/core-functions.sh || eerror "ERROR: /usr/lib/misc/core-functions.sh is not available."
-
-# cfg
-[ -f /etc/squashed-portage.conf ] && source /etc/squashed-portage.conf || eerror "ERROR: /etc/squashed-portage.conf is not available."
+extra_started_commands="sync"
 
 retuval() {
 	retval=${1}
@@ -100,15 +94,15 @@
 
 	for i in ${SQFSS[@]}; do
 		einfo "Mounting ${i}"
-		mount -rt squashfs -o loop,nodev,noexec "${SQFS_DIRNAME}/sqfs.${i}-current.sqfs" "${SQFS_DIRNAME}/${i}"
-		retuval ${?} "Error: mount -rt squashfs -o loop,nodev,noexec \"${SQFS_DIRNAME}/sqfs.${i}-current.sqfs\" \"${SQFS_DIRNAME}/${i}\""
+		mount -rt squashfs -o loop,nodev,noexec "${SQFS_DIRNAME}/sqfs.${i}-current.sqfs" "/usr/${i}"
+		retuval ${?} "Error: mount -rt squashfs -o loop,nodev,noexec \"${SQFS_DIRNAME}/sqfs.${i}-current.sqfs\" \"/usr/${i}\""
 		[ "${SQFS_DIST}" ] || \
-		if [ `echo ${DISTDIR} | grep "${SQFS_DIRNAME}/${i}"` ]; then
+		if [ `echo ${DISTDIR} | grep "/usr/${i}"` ]; then
 			mkdir -p /usr/local/distfiles
 			retuval ${?} "Error: mkdir -p /usr/local/distfiles"
 			mount -o bind "/usr/local/distfiles" "${DISTDIR}"
 			retuval ${?} "Error: mount -o bind /usr/local/distfiles ${DISTDIR}"
-			ewarn "DISTDIR is currently inside of ${SQFS_DIRNAME}/${i} tree. 
+			ewarn "DISTDIR is currently inside of /usr/${i} tree. 
 			It has been bind mounted to keep the SquashFS image small."
 		fi
 	done; unset i rw
@@ -128,8 +122,8 @@
 		[ ${a} ] && a=$[${a}+1] || local a=0
 		local RW=${FSRW[${a}]:-"/dev/shm/.${i}-rw"}
 		einfo "${RW}"
-		mount -t aufs -o "nodev,noexec,br:${RW}=rw:${SQFS_DIRNAME}/${i}=ro" aufs "${SQFS_DIRNAME}/${i}"
-		retuval ${?} "Error: mount -t aufs -o \"nodev,noexec,br:${RW}=rw:${SQFS_DIRNAME}/${i}=ro\" aufs \"${SQFS_DIRNAME}/${i}\""
+		mount -t aufs -o "nodev,noexec,br:${RW}=rw:/usr/${i}=ro" aufs "/usr/${i}"
+		retuval ${?} "Error: mount -t aufs -o \"nodev,noexec,br:${RW}=rw:/usr/${i}=ro\" aufs \"/usr/${i}\""
 		einfo "${SQFSS[${a}]} mounted in ${RW}"
 	done; unset a i rw
 	eend ${?}
@@ -167,10 +161,10 @@
 		for i in ${SQFSS[@]}; do
 		[ ${a} ] && a=$[${a}+1] || local a=0
 		local RW=${FSRW[${a}]:-"/dev/shm/.${i}-rw"}
-				umount -f -l -t aufs "${SQFS_DIRNAME}/${i}"
-		retuval ${?} "Error: umount -f -l -t aufs \"${SQFS_DIRNAME}/${i}\""
-		umount -f -l -t squashfs "${SQFS_DIRNAME}/${i}"
-		retuval ${?} "Error: umount -f -l -t squashfs \"${SQFS_DIRNAME}/${i}\""
+				umount -f -l -t aufs "/usr/${i}"
+		retuval ${?} "Error: umount -f -l -t aufs \"/usr/${i}\""
+		umount -f -l -t squashfs "/usr/${i}"
+		retuval ${?} "Error: umount -f -l -t squashfs \"/usr/${i}\""
 		/bin/rm -fr "${RW}"
 		retuval ${?} "Error: /bin/rm -fr \"${RW}\""
 		done; unset i retval
