diff --git a/Makefile b/Makefile
index 71b35b0..1a04aca 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-GEN=./dist/build/xcffibgen/xcffibgen
+GEN=xcffibgen
 AUTOPEP8=autopep8 --in-place --aggressive --aggressive
 
 XCBVER=$(shell pkg-config --modversion xcb-proto)
@@ -7,7 +7,7 @@ NCPUS=$(shell grep -c processor /proc/cpuinfo)
 PARALLEL=$(shell which parallel)
 
 # you should have xcb-proto installed to run this
-xcffib: $(GEN) module/*.py
+xcffib: module/*.py
 	$(GEN) --input $(XCBDIR) --output ./xcffib
 	cp ./module/*py ./xcffib/
 	sed -i "s/__xcb_proto_version__ = .*/__xcb_proto_version__ = \"${XCBVER}\"/" xcffib/__init__.py
