diff --git a/Makefile b/Makefile
index 2e8f035..cf85262 100644
--- a/Makefile
+++ b/Makefile
@@ -17,6 +17,10 @@ xcffib: module/*.py
 	cp ./module/*py ./xcffib/
 	sed -i "s/__xcb_proto_version__ = .*/__xcb_proto_version__ = \"${XCBVER}\"/" xcffib/__init__.py
 	@if [ "$(TRAVIS)" = true ]; then python xcffib/ffi_build.py; else python xcffib/ffi_build.py > /dev/null 2>&1 || python3 xcffib/ffi_build.py; fi
+	sed -i "s/version = .*/version = \"${ver}\"/" setup.py
+	sed -i "s/__version__ = .*/__version__ = \"${ver}\"/" xcffib/__init__.py
+	sed -r -i -e "s/(^version = \s*)[\"0-9\.]*/\1\"${ver}\"/" setup.py
+	sed -r -i -e "s/(^version:\s*)[0-9\.]*/\1${ver}/" xcffib.cabal
 
 .PHONY: xcffib-fmt
 xcffib-fmt: module/*.py
