diff --git a/Makefile b/Makefile
index 7a06b45..fae0757 100644
--- a/Makefile
+++ b/Makefile
@@ -13,11 +13,15 @@ GEN=$(CABAL) new-run exe:xcffibgen --
 
 # you should have xcb-proto installed to run this
 xcffib: module/*.py
-	$(GEN) --input $(XCBDIR) --output ./xcffib
+	dist/build/xcffibgen/xcffibgen --input $(XCBDIR) --output ./xcffib
 	cp ./module/*py ./xcffib/
 	touch ./xcffib/py.typed
 	sed -i "s/__xcb_proto_version__ = .*/__xcb_proto_version__ = \"${XCBVER}\"/" xcffib/__init__.py
 	@if [ "$(TRAVIS)" = true ]; then python xcffib/ffi_build.py; else python xcffib/ffi_build.py > /dev/null 2>&1 || python3 xcffib/ffi_build.py; fi
+	sed -i "s/version = .*/version = \"${ver}\"/" setup.py
+	sed -i "s/__version__ = .*/__version__ = \"${ver}\"/" xcffib/__init__.py
+	sed -r -i -e "s/(^version = \s*)[\"0-9\.]*/\1\"${ver}\"/" setup.py
+	sed -r -i -e "s/(^version:\s*)[0-9\.]*/\1${ver}/" xcffib.cabal
 
 .PHONY: xcffib-fmt
 xcffib-fmt: module/*.py
