--- language-python.cabal.orig	2014-08-11 21:25:16.493257467 -0500
+++ language-python.cabal	2014-08-11 21:24:37.102056260 -0500
@@ -30,11 +30,15 @@
    else 
       hs-source-dirs: src ghc_6_12-13
 
+   if (impl(ghc < 7.8))
+     build-depends: array == 0.4.*
+   else
+     build-depends: array == 0.5.*
+
    build-depends:
       base == 4.*,
       containers == 0.5.*,
       pretty == 1.1.*,
-      array == 0.4.*,
       transformers == 0.3.*,
       monads-tf == 0.1.*
    build-tools:      happy, alex
